### ISSUE TITLE
Possible fix for GNU extended filenames

### ISSUE THREAD

**Issue by vshymanskyy** [vidstige/ar#16]
Looks like this line:
https://github.com/vidstige/ar/blob/e0275f79281c1cc5d9f3f4622b8de3f76cf95a42/ar/archive.py#L78

Should be:
```py
end = data.find(b"\n", start)
```

After this fix, I was able to work with my files.
@vidstige please check it out

**Comment by vidstige** [comment]
Thanks for your contribution! But I think this will break tar files that does not have the `\n` character? Have you run the tests? I think maybe a better solution is to `.rstrip('\n')` characters from the filename instead to be compatible with even more tar-files.

**Comment by vidstige** [comment]
I just tested and this break Windows files and more.

**Comment by vidstige** [comment]
If you change something, please create a minimal tar-file that you can commit to the `test_data/` directory. I can do it for you if you tell me how you created this file. There is already a large coverage in there
* Windows lib-files
* Tar files produced on Mac
* Tar files produced on Linux
* Tar files produced on bsd

**Comment by vshymanskyy** [comment]
Thanks for checking it out.

Here's a simple example that reproduces the issue:
[reproduce.zip](https://github.com/user-attachments/files/16962833/reproduce.zip)

When you run `list_entries.py`, you'll notice that long filenames are parsed incorrectly, the first erroneous item would look like:
```log
Entry: lib_a-cephes_subr.o/
lib_a-cephes_subrf.o/
lib_a-cephes_subrl.o/
lib_a-copysignl.o/
lib_a-e_rem_pio2.o/
lib_a-e_remainder.o/
lib_a-ef_acosh.o/
lib_a-ef_atan2.o/
lib_a-ef_atanh.o/
lib_a-ef_hypot.o/
lib_a-ef_log10.o/
lib_a-ef_rem_pio2.o/
lib_a-ef_remainder.o/
lib_a-ef_scalb.o/
lib_a-el_hypot.o/
....
```

Note, this is  a single entry with a multi-line filename.
The expected name for this is entry is just `lib_a-cephes_subr.o`, this is why I submitted the fix proposal.
But obviously I missed something important. Maybe, we should look for `/` (0x2F) instead of `0x00`. The wiki says:
> System V ar uses a '/' character (0x2F) to mark the end of the filename

The `libm.a` that I've included with the test is the one shipped with `gcc-arm-none-eabi` package on Ubuntu and can be found in `/usr/lib/arm-none-eabi/newlib/thumb/v6-m/nofp/libm.a`

Other than that, the `ar` package works correctly and I'm able to use it to link real binaries
